nixos/modules/services/web-apps/peertube.nix

```diff
@@ -411,7 +411,9 @@ in {
         EOF
         ln -sf ${cfg.package}/config/default.yaml /var/lib/peertube/config/default.yaml
         ln -sf ${configFile} /var/lib/peertube/config/production.json
-        npm start
+        # Fails start when unable to write cache directory https://github.com/npm/cli/issues/4996
+        npm cache clean --cache /var/cache/peertube/.npm --force
+        npm start --cache /var/cache/peertube/.npm
```

Comment:
would you mind checking whether setting `npm_config_cache` works? 
```diff
diff --git a/nixos/modules/services/web-apps/peertube.nix b/nixos/modules/services/web-apps/peertube.nix
index 2fb05209c5c8..e0092a679be4 100644
--- a/nixos/modules/services/web-apps/peertube.nix
+++ b/nixos/modules/services/web-apps/peertube.nix
@@ -13,6 +13,7 @@ let
     NODE_EXTRA_CA_CERTS = "/etc/ssl/certs/ca-certificates.crt";
     NPM_CONFIG_PREFIX = cfg.package;
     HOME = cfg.package;
+    npm_config_cache = "/var/cache/peertube/.npm";
   };
 
   systemCallsList = [ "@cpu-emulation" "@debug" "@keyring" "@ipc" "@memlock" "@mount" "@obsolete" "@privileged" "@setuid" ];
```

Reply:
Yes it works.
Remove `npm cache clean --force` command?

Reply:
>Remove npm cache clean --force command?

yes, it should work without it